File: Frontend/Mobile/lib/models/community/comment.dart
class Comment {
  final int id;
  final String content;
  final String createdAt;
  final CommentChild? child;

  Comment({
    required this.id,
    required this.content,
    required this.createdAt,
    this.child,
  });

  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      id: json['id'],
      content: json['content'] ?? '',
      createdAt: json['createdAt'] ?? '',
      child: json['child'] != null
          ? CommentChild.fromJson(json['child'])
          : null,
    );
  }
}

class CommentChild {
  final int id;
  final String name;

  CommentChild({
    required this.id,
    required this.name,
  });

  factory CommentChild.fromJson(Map<String, dynamic> json) {
    return CommentChild(
      id: json['id'],
      name: json['name'] ?? '',
    );
  }
}

File: Backend/src/main/java/org/example/backend/model/community/Comment.java
package org.example.backend.model.community;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.example.backend.model.Child;
import org.example.backend.model.Parent;
import com.fasterxml.jackson.annotation.JsonIgnore;

import java.time.LocalDateTime;

@Entity
@Table(name = "comments")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(columnDefinition = "TEXT")
    private String content;

    private LocalDateTime createdAt;

    @ManyToOne
    @JoinColumn(name = "post_id")
    @JsonIgnore
    private Post post;

    @ManyToOne
    @JoinColumn(name = "child_id")
    private Child child;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "parent_id")
    private Parent parent;
}
